addons/test_mail/tests/test_mail_composer.py

```diff
@@ -32,6 +33,17 @@ def setUpClass(cls):
             name='Eglantine Employee', signature='--\nEglantine')
         cls.partner_employee_2 = cls.user_employee_2.partner_id
 
+        # User without the group "mail.group_mail_template_editor"
+        cls.user_rendering_restricted = mail_new_test_user(
+            cls.env, login='user_rendering_restricted',
+            groups='base.group_user',
+            company_id=cls.company_admin.id,
+            name='Code Template Restricted User',
+            notification_type='inbox',
+            signature='--\nErnest'
+        )
+        cls.env.ref('mail.group_mail_template_editor').users -= cls.user_rendering_restricted
```

Comment:
It's kind of hacky, here you create a user without the group "'mail.group_mail_template_editor" even if there's an imply "group_user => group_mail_template_editor" (it work but it shouldn't), so it might break in the future no ?

Isn't it better to set `mail.restrict.template.rendering` to True and to remove the group in the test itself ?

Reply:
> so it might break in the future no ?

That's why I added an `assertRaises` in the test. Changing the parameter for the whole file is maybe a bit more risky to break other tests, no?

Reply:
for me the best solution would be to create the user in the test (it's used only there in the file), with the config parameter set to True